File: 217-contains-duplicate/217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    let numCheck = {}
    for (let num of nums) {
        if (numCheck[num]) {
            return true
        } else {
            numCheck[num] = 1
        }
    }
    return false
};

File: 217-contains-duplicate/217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        int len = std::size(nums);
        map<int, int> numCheck;
        for (int i = 0; i < len; ++i)
        {
            if (numCheck.count(nums[i]))
            {
                return true;
            }
            else
            {
                numCheck[nums[i]] = 1;
            }
        }
        return false;
    }
};